### PR TITLE
libc/lib_utsname: Store version number for debugging and preventing optimization

### PR DESCRIPTION
**Pull request by W-M-R**
## Summary
1. In order to directly read the version information of the current file from elf or bin files, memory files, etc., for easy debugging and one-to-one correspondence.
## Impact
None
## Testing
None

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * Lacks specifics about the change. What functional part of the code is being modified? How does the change work? 
    * No mention of related issues or pull requests.

* **Impact:**
    * The impact section claims "None" across the board.  This is highly unlikely for any code change. Every change has *some* impact. Be specific. 
    * Consider:
        * Does this change add new functions or data structures?
        * Does it modify existing code paths?
        * Are there potential performance implications?

* **Testing:**
    * "None" is unacceptable.  Every PR needs evidence of testing. 
    * Provide details about your build host(s) and target(s) as specified in the requirements.
    * Include relevant testing logs demonstrating the issue before the change and the correct behavior after.

**Recommendations:**

1. **Elaborate on the Summary:** Clearly describe the technical details of your changes.
2. **Analyze the Impact:**  Thoroughly assess and document the impact of your changes, even if they seem minor. 
3. **Provide Testing Evidence:**  Demonstrate that your changes work as intended with specific testing details and logs. 
 
A well-documented PR makes it easier for maintainers to understand, review, and merge your contribution.

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
@W-M-R please copy the comment from Summary to the commit log message

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by W-M-R** [comment]
> @W-M-R please copy the comment from Summary to the commit log message

OK

File: libs/libc/misc/lib_utsname.c [hunk 1]
```diff
@@ -47,6 +47,20 @@
...
+ ****************************************************************************/
+
+static char g_sysname[] = "NuttX";
+static char g_machine[] = CONFIG_ARCH;
+static char g_release[] = CONFIG_VERSION_STRING;
+
+#if defined(__DATE__) && defined(__TIME__)
+static char g_version[] = CONFIG_VERSION_BUILD " " __DATE__ " " __TIME__;
+#else
+static char g_version[] = CONFIG_VERSION_BUILD;
```
Comment (anchao) [inline libs/libc/misc/lib_utsname.c:61]:
```suggestion
static const char g_version[] = CONFIG_VERSION_BUILD;
```

**Review by anchao — CHANGES_REQUESTED** [review]

File: libs/libc/misc/lib_utsname.c (hunk 1 above)
Comment (W-M-R) [reply libs/libc/misc/lib_utsname.c:61]: These data do not want to be read-only, they need to exist in memory, so they need to be located in the data segment. For example, if we want to match which elf all exported memory files correspond to, we need to read out the version and other information from the memory

Comment (anchao) [reply libs/libc/misc/lib_utsname.c:61]: seems may be a special case? could we just change g_version to data and change other variables back to rodata?

File: libs/libc/misc/lib_utsname.c [hunk 2]
```diff
@@ -51,7 +51,7 @@
  * Private Data
  ****************************************************************************/
 
-#if defined(__DATE__) && defined(__TIME__) \
+#if defined(__DATE__) && defined(__TIME__) && \
```
Comment (xiaoxiang781216) [inline libs/libc/misc/lib_utsname.c:54]: squash to previous one

**Review by anchao — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**